homeassistant/components/wallbox/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""The Wallbox integration."""
+import asyncio
+import logging
+
+import requests
+from wallbox import Wallbox
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_CONNECTIONS, CONF_STATION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+class WallboxHub:
+    """Wallbox Hub class."""
+
+    def __init__(self, station, username, password):
+        """Initialize."""
+        self._station = station
+        self._username = username
+        self._password = password
+
+    async def async_authenticate(self, hass) -> bool:
+        """Authenticate using Wallbox API."""
+        try:
+            wallbox = Wallbox(self._username, self._password)
+            await hass.async_add_executor_job(wallbox.authenticate)
+            return True
+        except requests.exceptions.HTTPError as wallbox_connection_error:
+            if wallbox_connection_error.response.status_code == 403:
+                raise InvalidAuth from wallbox_connection_error
+            raise ConnectionError from wallbox_connection_error
+
+    async def async_get_data(self, hass) -> bool:
+        """Get new sensor data for Wallbox component."""
+        try:
+            wallbox = Wallbox(self._username, self._password)
+            await hass.async_add_executor_job(wallbox.authenticate)
+            data = await hass.async_add_executor_job(
+                wallbox.getChargerStatus, self._station
+            )
+            return data
+        except requests.exceptions.HTTPError as wallbox_connection_error:
+            if wallbox_connection_error.response.status_code == 403:
+                raise InvalidAuth from wallbox_connection_error
+            raise ConnectionError from wallbox_connection_error
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Wallbox component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Wallbox from a config entry."""
+    wallbox = WallboxHub(
```

Comment:
```suggestion
    self.wallbox = WallboxHub(
```
why not reuse ?

Reply:
that would not work there, its not inside the class. Do you mean something different?

Reply:
ups my bad